images/build.sh

```diff
@@ -2,9 +2,60 @@
 
 set -eu
 
-source docker-version.sh
-docker image build --build-arg KAFKA_VERSION=${KAFKA_BASE_TECH_VERSION} -t mesosphere/kafka:${DOCKER_IMAGE_VERSION}-${KAFKA_BASE_TECH_VERSION} .
+function print_help() {
+  echo "Usage: ${0} [OPTIONS]"
+  echo
+  echo "OPTIONS:"
+  echo "--push             push the recently build image"
+  echo "--image            docker image to build. valid options are '-image=kafka' and '-image=cruise-control'"
+  echo
+}
 
-if [[ $1 == "push" ]]; then
-    docker push mesosphere/kafka:${DOCKER_IMAGE_VERSION}-${KAFKA_BASE_TECH_VERSION}
+PUSH_IMAGE="false"
+for arg in "$@"
+do
+    case $arg in
+        -p|--push)
+        PUSH_IMAGE="true"
+        shift
+        ;;
+        -i=*|--image=*)
+        IMAGE_NAME="${arg#*=}"
+        shift
+        ;;
+        *)
+        print_help
+        exit 1
+        ;;
+    esac
+done
+
+if [[ -z "${IMAGE_NAME:-}" ]]; then
+  print_help
+fi
+
+if [[ "${IMAGE_NAME}" == "kafka" ]]; then
+  source kafka/docker-version.sh
+  docker image build --build-arg KAFKA_VERSION=${KAFKA_BASE_TECH_VERSION} -t mesosphere/kafka:${KAFKA_DOCKER_IMAGE_VERSION}-${KAFKA_BASE_TECH_VERSION} ./kafka
+elif [[ "${IMAGE_NAME}" == "cruise-control" ]]; then
+  source cruise-control/docker-version.sh
+  docker image build --build-arg CRUISE_CONTROL_VERSION=${CRUISE_CONTROL_VERSION} --build-arg CRUISE_CONTROL_UI_VERSION=${CRUISE_CONTROL_UI_VERSION} \
+    -t mesosphere/cruise-control:${CRUISE_CONTROL_DOCKER_IMAGE_VERSION} ./cruise-control
```

Comment:
Why the different versioning scheme for cruise-control image (no base tech version)?

Reply:
yes, this in application and we shouldn't add complexity to this app, unless we are providing an operator for cruise-control itself. 
I would have used something existing and official docker image but as there is no official docker image for cruise-control, went with this one

Reply:
KUDO C* [does something similar](https://github.com/mesosphere/kudo-cassandra-operator/blob/4517807137461f3308d82cf68bdd619ec9e50bc6/metadata.sh#L48-L62).

The Cassandra image tag ends up being something like `3.11.5-0.1.1` (`3.11.5` is the Cassandra version) and the Prometheus exporter image tag ends up being something like `2.2.1-0.1.1` (`2.2.1` is the Prometheus exporter version).

Reply:
let's use the same pattern here also then: https://github.com/mesosphere/kudo-kafka-operator/pull/29/commits/092e3b824e69b93e8ed04b781805f07c7a208cc2